frontstage/views/account/accept_transfer_survey.py

```diff
@@ -0,0 +1,119 @@
+    try:
+        response = party_controller.verify_pending_survey_token(token)
+        pending_transfer_surveys = response.json()
+        transfer_dict = {}
+        distinct_businesses = set()
+        batch_number = pending_transfer_surveys[0]['batch_no']
+        originator_party = party_controller.get_respondent_party_by_id(pending_transfer_surveys[0]['shared_by'])
+        transferred_by = originator_party['emailAddress']
+        for pending_transfer_survey in pending_transfer_surveys:
+            distinct_businesses.add(pending_transfer_survey['business_id'])
+        for business_id in distinct_businesses:
+            business_surveys = []
+            for pending_transfer_survey in pending_transfer_surveys:
+                if pending_transfer_survey['business_id'] == business_id:
+                    business_surveys.append(survey_controller.get_survey(app.config['SURVEY_URL'],
+                                                                         app.config['BASIC_AUTH'],
+                                                                         pending_transfer_survey['survey_id']))
+            selected_business = get_business_by_id(business_id)
+            transfer_dict[selected_business[0]['name']] = {
+                'trading_as': selected_business[0]['trading_as'],
+                'surveys': business_surveys
+            }
+        return render_template('surveys/surveys-transfer/summary.html',
+                               transfer_dict=transfer_dict,
+                               batch_no=batch_number,
+                               transferred_by=transferred_by)
+
+    except ApiError as exc:
+        # Handle api errors
+        if exc.status_code == 409:
+            logger.info('Expired transfer survey email verification token', token=token, api_url=exc.url,
+                        api_status_code=exc.status_code)
+            abort(409)
+        elif exc.status_code == 404:
+            logger.warning('Unrecognised transfer survey email verification token', token=token, api_url=exc.url,
+                           api_status_code=exc.status_code)
+            abort(404)
+        else:
+            logger.info('Failed to verify transfer survey email', token=token, api_url=exc.url,
+                        api_status_code=exc.status_code)
+            raise exc
+
+
+@account_bp.route('/confirm-transfer-surveys/<batch>', methods=['GET'])
+def accept_transfer_surveys(batch):
+    """
+    Accept endpoint when a transfer survey summary is accepted
+    :param batch: batch number
+    :type batch: str
+    """
+    logger.info('Attempting to get batch number', batch_number=batch)
+    try:
+        response = party_controller.get_pending_surveys_batch_number(batch)
+        is_existing_user = _is_existing_account(response.json()[0]['email_address'])
+    except ApiError as exc:
+        logger.error('Failed to confirm transfer survey', status=exc.status_code, batch_number=batch)
```

Comment:
This might be a more acceptable time to throw a generic 500, but it's worth considering if you want to feed anything back here

Reply:
🎨

Reply:
I think it will be fed back as we are raising the error. Please correct me if I am wrong.

Reply:
I checked with Gareth and Jeremy, generic frontstage 500 error is ok.